bpf/bpf_lxc.c

```diff
@@ -1192,6 +1198,20 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		key.family = ENDPOINT_KEY_IPV4;
 		key.cluster_id = (__u8)cluster_id;
 
+#if !defined(ENABLE_NODEPORT) && defined(ENABLE_HOST_FIREWALL)
+		/*
+		 * For the host firewall, traffic from a pod to a remote node is sent
+		 * through the tunnel. In the case of node to remote pod traffic via
+		 * externalTrafficPolicy=Local services, packets may be DNATed when
+		 * they enter the remote node (without being SNATed at the same time).
+		 * If kube-proxy is used, the response needs to go through the stack
+		 * to apply the correct reverse DNAT, and then be routed accordingly.
+		 * See #14674 for details.
+		 */
+		if (ct_status == CT_REPLY && identity_is_remote_node(*dst_sec_identity))
```

Comment:
drive-by - strictly speaking, I believe we might also need to handle `CT_RELATED` here:
https://github.com/cilium/cilium/blob/a9d83fe157e14bfa10a1f3437bc0c4f98a5c800f/bpf/lib/conntrack.h#L823-L827

And yes, wouldn't it be nice if we had a `ct_result_is_reply()` helper for such cases :slightly_smiling_face: .

Reply:
TIL! Thanks, fixed.